### PR TITLE
Able to change image and text

### LINKED ISSUE #93: request bmp image file support right color
screenshot

### LINKED ISSUE #97: Got color channel bug on macOS
Got color channel bug on macOS
technology screenshot

_Originally posted by @ntut-rick in https://github.com/ntut-open-source-club/practical-tools-for-simple-design/issues/96#issuecomment-1932068671_

### PR DESCRIPTION
**Pull request by NOOBDY**
Changelog:
- Added `Image::SetImage()`
- Added `Text::SetText()` and `Text::SetColor()`
- Fixed bug where color channels are inverted(i.e. blue text shows up red)
- Change `GameObject.m_Drawable` to `shared_ptr` make it changeable

Note:
Currently loading the same image still isn't cached

Related Issues:
- Fix #93 
- Fix #97 
- #64

### CODE REVIEW

**Comment by ntut-rick** [comment]
Got color channel bug on macOS
technology screenshot

**Review by nonorze-VOXX — APPROVED** [review]
LGTM IWOMM

**Comment by ntut-rick** [comment]
Tested on macOS with PNG, BMP
Tested on windows with PNG, BMP

**Review by ntut-rick — DISMISSED** [review]

**Comment by ntut-rick** [comment]
I still believe that duplicated code in `Image` and `Text` can be consolidated. Nothing else

**Comment by NOOBDY** [comment]
technically you can, but I haven't come up with a way to do it that makes sense semantically

File: src/Core/TextureUtils.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    case SDL_PIXELFORMAT_BGR24:
+        return GL_BGR;
+    case SDL_PIXELFORMAT_XRGB8888:
+        return GL_BGRA;
+    case SDL_PIXELFORMAT_XBGR8888:
+        return GL_RGBA;
+    case SDL_PIXELFORMAT_ARGB8888:
+        return GL_BGRA;
+    case SDL_PIXELFORMAT_ABGR8888:
+        return GL_RGBA;
```
[the remark this answers is not in the document]
Comment (NOOBDY) [reply src/Core/TextureUtils.cpp:17]: I don't understand what you are saying

[the remark this answers is not in the document]
Comment (NOOBDY) [reply src/Core/TextureUtils.cpp:17]: if you `return` there is no need to `break`

Comment (Dragonfly911117) [inline src/Core/TextureUtils.cpp:17]:
```suggestion
    switch (format) {
    case SDL_PIXELFORMAT_RGB24:
        return GL_RGB;
    case SDL_PIXELFORMAT_BGR24:
        return GL_BGR;
    case SDL_PIXELFORMAT_XRGB8888:
    case SDL_PIXELFORMAT_ARGB8888:
        return GL_BGRA;
    case SDL_PIXELFORMAT_XBGR8888:
    case SDL_PIXELFORMAT_ABGR8888:
        return GL_RGBA;
```

**Review by Dragonfly911117 — APPROVED** [review]

**Pull request merged**